sechub-schedule/src/main/java/com/daimler/sechub/domain/schedule/strategy/SchedulerStrategyFactory.java

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: MIT
+package com.daimler.sechub.domain.schedule.strategy;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SchedulerStrategyFactory {
+    
+    private static final Logger LOG = LoggerFactory.getLogger(SchedulerStrategyFactory.class);
+    
+    @Autowired
+    FirstComeFirstServeSchedulerStrategy fifoStrategy;
+    
+    @Autowired
+    OnlyOneScanPerProjectAtSameTimeStrategy oosppStrategy;
+
+    @Value("${sechub.scheduler.strategy.id:}")
+    private String strategyId;
+
+    public SchedulerStrategy build() {
+        
+        SchedulerStrategyId strategy = SchedulerStrategyId.getId(strategyId);
+                
+        if (strategy == null) {
```

Comment:
I would prefer I you would not use a guard close here, but do following.
1. rename strategy to strategyId
2. introduce a static constant FALLBACK_SCHEDULER_STRATEGY=SchedulerStrategyId.FirstComeFirstServer
2. do following here:
```java
if (stategyId==null){
     strategyId=FALLBACK_SCHEDULER_STRATEGY;
     LOG.info("SCHEDULER STRATEGY : NULL. Defaulting to  {}",  strategyId);
}
```
(so just revertiy to default strategyId, and complete workflow (loging switch etc.) will be done as in any other case )